### ISSUE TITLE
VLA-JEPA Lab 01: best freeze-Qwen BC 10K is 1/10; WM, extra steps, and unfreeze-Qwen all worse

### ISSUE THREAD

**Issue by alexhegit** [rocPAI-Forge/so101-simstudio#7]
## Summary

Lab 01 VLA-JEPA from [`lerobot/VLA-JEPA-LIBERO`](https://huggingface.co/lerobot/VLA-JEPA-LIBERO) on the **same 50-episode** joint-position dataset as ACT / MolmoAct2 is a **closed joint-space LIBERO→SO-101 transfer failure**, not evidence that pick-and-place is too hard. Same data: ACT 6-D **~58%** and MolmoAct2 6-D **~54%** full-range.

**Keep only freeze-Qwen BC 10K** (Hub artifact). Best closed-loop is **1/10** fixed-pose EGL. World-model 20K, BC 20K, and unfreeze-Qwen 10K all scored **0/10**. Classroom policy stays **ACT 6-D / MolmoAct2 6-D**.

**Hub (negative-result artifact):** https://huggingface.co/alexhegit/so101-simstudio-lab01-pnp-vla-jepa  
**Ledger:** Lab 01 [§6.4](https://github.com/rocPAI-Forge/so101-simstudio/blob/main/labs/lab01_pnp/lab01_pnp.md) / [§6.4.2](https://github.com/rocPAI-Forge/so101-simstudio/blob/main/labs/lab01_pnp/lab01_pnp.md)

Sibling failures: GR00T N1.7 [issue #6](https://github.com/rocPAI-Forge/so101-simstudio/issues/6) (approach yes / grasp no). FastWAM and LingBot-VA get their own issues.

## Failure data (closed-loop)

Fixed cube `(0.27, 0.20, −8°)`, `reset_arm: home`, sync, unless noted. Cameras: `top→image`, `wrist→image2` (drop `front`).

| Recipe | Protocol | Backend | Success | Funnel / symptom |
|--------|----------|---------|---------|------------------|
| **Freeze-Qwen BC 10K** (chunk 30, WM off) | Demo-mean `home_joints` | EGL | **1/10** | Approach; grasp often fails |
| same weights | Keyboard `home` | EGL | **1/10** | same |
| same weights | Demo-mean | GLFW | **0/10** | Grasp-stage failures |
| same + gripper proximity oracle | Demo-mean | EGL | **0/10** | Close every ep; never lift; ~4 cm Y miss |
| WM 20K (chunk 7, Qwen trainable, WM 0.1) | Keyboard `home` | EGL | **0/10** | Cube never leaves spawn |
| Same BC **20K** | demo-mean **and** keyboard home | EGL | **0/10** | Worse than 10K |
| Action-head-only 3K from BC 10K | Demo-mean / keyboard | EGL | 1/10 / 0/10 | No gain vs BC 10K |
| **Unfreeze Qwen 10K** from BC 010000 | Demo-mean | OSMesa | **0/10** | `reach_8cm=10/10 → near_4cm=6/10 → close=0/10` |
| Unfreeze Qwen 10K | Keyboard home | OSMesa | **0/10** | `near_4cm=0/10` |
| Matched n=3 audit | Demo-mean (causal set) | — | BC **0/3** place | BC: lift then **drop before box**; unfreeze: **close_after_open=0/3** |

Open-loop vs ACT: ~5× worse RMSE; does not pass start RMSE `<0.08` / cube–action `r>0.5`.

Do **not** Hub: WM 20K, BC 20K, unfreeze-Qwen. Unfreeze lives on DORobot: `/mnt/doscratch/outputs/train/lab01_pnp_vla_jepa_state6_bc_unfreeze/`.

## What worked (do not undo)

- 6-D joint `.pos` (`LAB01_STATE_DIM=6`); `binarize_gripper_action=false` (Lab 01 gripper is not two modes).
- Freeze Qwen + WM off is the **least-bad** recipe on this set.
- Reinit action/state heads for SO-101 (`reinit_modules`). LIBERO is 7-DoF / 8-D state.
- ROCm extras via `scripts/install-lerobot-extra-deps.sh`. **Never** `uv pip install 'lerobot[vla_jepa]'` (or similar) inside `.venv-rocm` if that pulls CUDA torch.
- `eval.py` keeps fine-tune `state_dim=6` even if the LIBERO base lists 8-D `input_features`.

## Failed recipes (do not repeat as first levers)

1. **World-model 20K** (`enable_world_model`, weight 0.1, chunk 7) — cube sensitivity **worse** than BC; predictor is not called by `predict_action`.
2. **More BC steps on frozen Qwen (20K)** — 0/10, worse than 10K.
3. **Unfreeze Qwen on 50 episodes** — representation drift: action head fitted to a **fixed** feature space; thawing ~2.3B params moved features; close disappeared first. Train loss ~0.011 did not help.
4. **Action-head-only 3K**, shorter chunks (10 / 5), gripper threshold snap, **gripper oracle** — none recovered a centered grasp/lift.
5. Treating 50 SO-101 demos as equivalent to the paper’s 220K human + 76K robot + ~2K LIBERO fine-tune.

## Lessons

1. **Death is at / after grasp** on the keeper (approach works; jaws often stay open ~0.85–0.93 rad, fingertip ~4 cm off in Y). After unfreeze, death moves **before close**.
2. This is **LIBERO vision/action-head transfer on 50 episodes**, not “Lab 01 is too hard” (ACT/Molmo succeed on the same labels).
3. **Unfreezing the VLM is not the first lever** on this set (same family as GR00T vis-unfreeze).
4. Camera rename (`top→image`) is key-name alignment only; top-down ≠ LIBERO agentview. Second-order vs the action/vision transfer.

## Next work (tracked here)

- [ ] Do **not** open further Lab 01 long trains (full unfreeze, more BC, WM weight sweeps) aimed at classroom success on **this** 50-ep set.
- [ ] If revisiting JEPA: **more same-embodiment grasp→place data**, keep freeze-Qwen BC 10K as the baseline to beat, re-eval the §6.4 funnel (`reach → near → close → lift → over box → place`).
- [ ] Optional: paper-scale / same-embodiment pretrain — out of Lab 01 classroom path.
- [ ] Classroom demos stay ACT 6-D / MolmoAct2 6-D until `k/n` is non-trivial on the Lab 01 protocol.

## Eval

```bash
LAB01_POLICY_PATH=./outputs/hub/lab01_pnp_vla_jepa \
LAB01_EVAL_CONFIG=labs/lab01_pnp/configs/rollout_vla_jepa_demo_start.yaml \
LAB01_N_ACTION_STEPS=30 LAB01_EVAL_EPISODES=10 \
LAB01_MUJOCO_GL=egl LAB01_RENDER_WINDOW=false \
  ./labs/lab01_pnp/eval.cmd
```
